File: beginner/1049.c
# include <stdio.h>
# include <string.h>

int main(){
	char primeiraPalavra[15] = {""};
	char segundaPalavra[15] = {""};
	char terceiraPalavra[15] = {""};
	int retorno = 1;
	
	scanf("%s", primeiraPalavra);
	scanf("%s", segundaPalavra);
	scanf("%s", terceiraPalavra);
	
	retorno = strcmp(primeiraPalavra, "vertebrado");
	if( retorno == 0 ){
		retorno = strcmp(segundaPalavra, "ave");
		if(  retorno == 0 ){
			retorno = strcmp(terceiraPalavra, "carnivoro");
			if(  retorno == 0 ){
				printf("aguia\n");
			}
			retorno = strcmp(terceiraPalavra, "onivoro");
			if(  retorno == 0 ){
				printf("pomba\n");
			}
		}
		retorno = strcmp(segundaPalavra, "mamifero");
		if(  retorno == 0 ){
			retorno = strcmp(terceiraPalavra, "onivoro");
			if(  retorno == 0 ){
				printf("homem\n");
			}
			retorno = strcmp(terceiraPalavra, "herbivoro");
			if( retorno == 0){
				printf("vaca\n");
			}
		}
	}
	
	retorno = strcmp(primeiraPalavra, "invertebrado");
	if( retorno == 0){
		retorno = strcmp(segundaPalavra, "inseto");
		if( retorno == 0){
			retorno = strcmp(terceiraPalavra, "hematofago");
			if( retorno == 0){
				printf("pulga\n");
			}
			retorno = strcmp(terceiraPalavra, "herbivoro");
			if( retorno == 0){
				printf("lagarta\n");
			}
		}
		
		retorno = strcmp(segundaPalavra, "anelideo");
		if( retorno == 0){
			retorno = strcmp(terceiraPalavra, "hematofago");
			if( retorno == 0){
				printf("sanguessuga\n");
			}
			retorno = strcmp(terceiraPalavra, "onivoro");
			if( retorno == 0){
				printf("minhoca\n");
			}
		}
	}
	
	return 0;
}

File: beginner/1049.py
primeiraPalavra = input()
segundaPalavra = input()
terceiraPalavra = input()

if primeiraPalavra == "vertebrado":
	if segundaPalavra == "ave":
		if terceiraPalavra == "carnivoro":
			print("aguia")
		if terceiraPalavra == "onivoro":
			print("pomba")
	if segundaPalavra == "mamifero":
		if terceiraPalavra == "onivoro":
			print("homem")
		if terceiraPalavra == "herbivoro":
			print("vaca")
if primeiraPalavra == "invertebrado":
	if segundaPalavra == "inseto":
		if terceiraPalavra == "hematofago":
			print("pulga")
		if terceiraPalavra == "herbivoro":
			print("lagarta")
	if segundaPalavra == "anelideo":
		if terceiraPalavra == "hematofago":
			print("sanguessuga")
		if terceiraPalavra == "onivoro":
			print("minhoca")
